SyncTelemetry/SyncTelemetryEvents.swift

```diff
@@ -12,42 +12,74 @@ public let PrefKeySyncEvents = "sync.telemetry.events"
 public typealias IdentifierString = String
 public extension IdentifierString {
     func validate() -> Bool {
-        // Regex located here: http://gecko.readthedocs.io/en/latest/toolkit/components/telemetry/telemetry/collection/events.html#limits
         let regex = try! NSRegularExpression(pattern: "^[a-zA-Z][a-zA-Z0-9_.]*[a-zA-Z0-9]$", options: [])
         return regex.matches(in: self, options: [], range: NSRange(location: 0, length: self.count)).count > 0
     }
 }
 
-// Telemetry Events
-// Documentation: http://gecko.readthedocs.io/en/latest/toolkit/components/telemetry/telemetry/collection/events.html#events
-public struct Event {
+public struct Event: Decodable {
     let timestamp: Timestamp
     let category: IdentifierString
     let method: IdentifierString
     let object: IdentifierString
+
     let value: String?
     let extra: [String: String]?
 
-    public init(category: IdentifierString,
-                method: IdentifierString,
-                object: IdentifierString,
-                value: String? = nil,
-                extra: [String: String]? = nil) {
-
-        self.init(timestamp: .uptimeInMilliseconds(),
-                  category: category,
-                  method: method,
-                  object: object,
-                  value: value,
-                  extra: extra)
+    public init(
```

Comment:
Could you review the indentation in this file? We use the default Xcode indentation with 4 spaces. This can be easily formatted using `Ctrl + i` shortcut in Xcode

Reply:
Sure, I'll get this reformatted. I typically use AppCode and the default reformatting is obviously different.